aspnetcore/fundamentals/routing.md

```diff
@@ -1,77 +1,326 @@
+
+The methods provided by `LinkGenerator` support standard link generation capabilities for any type of address. The most convenient way to use the link generator is through extension methods that perform operations for a specific address type.
+
+| Extension Method   | Description                                                         |
+| ------------------ | ------------------------------------------------------------------- |
+| `GetPathByAddress` | Generates a URI with an absolute path based on the provided values. |
+| `GetUriByAddress`  | Generates an absolute URI based on the provided values.             |
+
+> [!WARNING]
+> Pay attention to the following security implications of calling `LinkGenerator` methods:
+>
+> Use `GetUri*` extension methods with caution in an app configuration that doesn't validate the `Host` header of incoming requests. If the `Host` header of incoming requests isn't validated, untrusted request input can be sent back to the client in URIs in a view/page. We recommend that all production apps configure their server to validate the `Host` header against known valid values.
+>
+> Use `LinkGenerator` with caution in middleware in combination with `Map` or `MapWhen`. `Map*` changes the base path of the executing request, which affects the output of link generation. All of the `LinkGenerator` APIs allow specifying a base path. Always specify an empty base path to undo `Map*`'s affect on link generation.
+
+A few differences exist between endpoint routing in ASP.NET Core 2.2 or later and earlier versions of routing in ASP.NET Core:
+
+* The endpoint routing system doesn't support `IRouter`-based extensibility, including inheriting from `Route`.
+
+* Endpoint routing doesn't support [WebApiCompatShim](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.WebApiCompatShim). Use the 2.1 [compatibility version](xref:mvc/compatibility-version) (`.SetCompatibilityVersion(CompatibilityVersion.Version_2_1)`) to continue using the compatibility shim.
+
+* Endpoint Routing has different behavior for the casing of generated URIs when using conventional routes.
+
+  Consider the following default route template:
+
+  ```csharp
+  app.UseMvc(routes =>
+  {
+      routes.MapRoute("default", "{controller=Home}/{action=Index}/{id?}");
+  });
+  ```
+
+  Suppose you generate a link to an action using the following route:
+
+  ```csharp
+  var link = Url.Action("ReadPost", "blog", new { id = 17, });
+  ```
+
+  With `IRouter`-based routing, this code generates a URI of `/blog/ReadPost/17`, which respects the casing of the provided route value. Endpoint routing in ASP.NET Core 2.2 or later produces `/Blog/ReadPost/17` ("Blog" is capitalized). Endpoint routing provides a parameter transformer interface called `IOutboundParameterTransformer` that can be used to customize this behavior globally or to apply different conventions for mapping URLs.
```

Comment:
`provides a parameter transformer interface` -> `provides a feature called`

Reply:
I can't use the word "feature" .... it's been **_banned_** from topics. :skull: lol

I'll go with ...

> Endpoint routing provides the `IOutboundParameterTransformer` interface that ...

Reply:
*Yikes!* ... I'm glad you said something. I found ~4 more "features" instances in the topic. They don't just **_release you_** for using banned words ... it's *death by firing squad!* ... :boy: :gun: :smile: lol